### PR TITLE
Make union/except/intersect chainable as SelectManager

### PR DESCRIPTION
**Pull request by jsanders**
Fixes #118.

This is the same as #239, but against master instead of 4-0-stable. It would be really nice to get this in so we don't need to keep freshening it! Or if there's a reason it isn't being merged, feedback would be great.

### CODE REVIEW

**Comment by amalagaura** [comment]
It is frustrating AREL doesn't support this, there are a lot of use cases for chaining union operators. It would be great if this was resolved.

**Comment by jsanders** [comment]
@amalagaura: Yeah, I rely on this behavior and keep needing to update this branch to work with new versions. If the maintainers have no interest in merging this functionality, perhaps I should instead figure out a way to do it with a monkey-patch or a separate library.

@rafaelfranca / @tenderlove / other maintainers: I'm not sure if you've seen this, but a thumbs-up or thumbs-down on this functionality would be really nice, so that I have a sense for whether or not I'm wasting my time on it. If the functionality is desired, but you don't like the implementation, I'm willing to put together a different PR that speaks to that.

**Comment by amalagaura** [comment]
This seems like a no-brainer. I also have 3 Arel::SelectManagers that I need to merge and I am forced to do a .to_sql  and join(' UNION '). 

Arel needs union to not return a Node::Binary and instead return a proper SelectManager, otherwise union really is broken since SQL definitely supports chainable unions.

**Comment by sockmonk** [comment]
I'd also like to see a better way to work with unions. Feedback on @jsanders  pull request please?

**Comment by ClayShentrup** [comment]
How has this been ignored a year? @tenderlove ??

**Comment by tenderlove** [comment]
I sincerely apologize for not having time for this.  Can you please rebase it and I will review.  I'm really sorry.

**Comment by jsanders** [comment]
@tenderlove: No worries! I'll find some time to rebase and ping you again.

**Comment by ryanswood** [comment]
@jsanders Did you rebase? We'd love to have this. Thanks!

**Comment by jsanders** [comment]
@tenderlove: Rebased this onto current master and things look pretty good. Feedback welcome!

Sorry for the delay, folks.

File: lib/arel/select_manager.rb [hunk 1]
```diff
@@ -6,11 +6,15 @@ class SelectManager < Arel::TreeManager
...
     STRING_OR_SYMBOL_CLASS = [Symbol, String]
 
-    def initialize engine, table = nil
+    def initialize engine, table = nil, cores = nil
       super(engine)
-      @ast   = Nodes::SelectStatement.new
-      @ctx    = @ast.cores.last
-      from table
+      if cores
+        @ast = Nodes::SelectStatement.new cores
```
Comment (tamird) [inline lib/arel/select_manager.rb]: :( guess this was ignored

Comment (jsanders) [inline lib/arel/select_manager.rb]: Actually, I did it before, but ended up skipping it when I rebased when it caused a conflict and I decided it was pretty trivial. I guess if people feel strongly, I can add it back.

**Comment by ryanswood** [comment]
@tenderlove Waiting on your tender PR love. Thanks.

**Comment by amalagaura** [comment]
@tenderlove Any updates? I have another use case and don't want to use `join(' OR ')`

**Comment by nilcolor** [comment]
Hello. What about this?
```ruby
Arel::Nodes::Union.new(select_manager_1, Arel::Nodes::Union.new(select_manager_2, select_manager_3))
```
This gives us "chained" unions. Some over-parenthesized but workable solution. We can generalize it a'lil:
```ruby
def build_union(left, right)
  if right.length > 1
    Arel::Nodes::UnionAll.new(left, build_union(right[0], right[1..-1]))
  else
    Arel::Nodes::UnionAll.new(left, right[0])
  end
end

managers = [select_manager_1, select_manager_2, select_manager_3]
build_union(managers[0], managers[1..-1]).to_sql
# => ( (SELECT table1.* from table1)
#    UNION ALL
#    ( (SELECT table2.* from table2)
#    UNION ALL
#    (SELECT table3.* from table3) ) )
```
This works in Arel 4.0.2... No need to `join` lots of `to_sql`...

**Comment by matthewd** [comment]
Per #523, Arel development is moving to [rails/rails](https://github.com/rails/rails).

If this PR is still relevant, please consider reopening it over there.

**Pull request closed without merging**